hie-plugin-api/Haskell/Ide/Engine/GhcModuleCache.hs

```diff
@@ -100,21 +102,59 @@ data LookupCradleResult = ReuseCradle | LoadCradle CachedCradle | NewCradle File
 lookupCradle :: FilePath -> GhcModuleCache -> LookupCradleResult
 lookupCradle fp gmc =
   case currentCradle gmc of
-    Just (dirs, _c) | (any (\d -> d `isPrefixOf` fp) dirs) -> ReuseCradle
+    Just (dirs, _c, _) | any (`isPrefixOf` fp) dirs -> ReuseCradle
     _ -> case T.match (cradleCache gmc) (B.pack fp) of
            Just (_k, c, _suf) -> LoadCradle c
            Nothing  -> NewCradle fp
 
-data CachedCradle = CachedCradle BIOS.Cradle HscEnv
+getComponentOptions
+  :: (MonadIO m, HasGhcModuleCache m)
+  => FilePath
+  -> m (Maybe BIOS.ComponentOptions)
+getComponentOptions fp = do
+  mc <- getModuleCache
+  case currentCradle mc of
+    Just (dirs, cradle, co) | any (`isPrefixOf` fp) dirs -> case co of
```

Comment:
The `dirs` here are always absolute paths right?

Reply:
Yes they do!